src/TextInput/TextInput.svelte

```diff
@@ -86,6 +114,8 @@
   /** @type {(e: Event) => void} */
   const onInput = (e) => {
     value = parse(e.target.value);
+    if (maxCount && value.length > maxCount) value = value.slice(0, maxCount);
+    if (counter && maxCount) updateCount();
     dispatch("input", value);
```

Comment:
Since other events were modified, I recommend forwarding `on:input` on the
`<input>` element and removing this dispatched `input`. It's a breaking change,
but one we'll get around to anyway.